assets/javascripts/api-initializers/extend-for-survey.gjs

```diff
@@ -0,0 +1,44 @@
+import routeAction from "discourse/helpers/route-action";
+import { apiInitializer } from "discourse/lib/api";
+import Survey from "../components/surveys/survey";
+
+const SurveyShim = <template>
+  <Survey
+    @surveyName={{@data.surveyName}}
+    @post={{@data.post}}
+    @showLogin={{routeAction "showLogin"}}
+  />
+</template>;
+
+export default apiInitializer((api) => {
+  api.modifyClass("controller:topic", {
+    pluginId: "discourse-surveys",
+    subscribe() {
+      this._super(...arguments);
+      this.messageBus.subscribe("/surveys/" + this.get("model.id"), (msg) => {
```

Comment:
anonymous function work correctly for subscribe/unsubscribe?

Reply:
In this case yes, because the `.unsubscribe` call is removing all subscriptions from the messagebus channel.

Reply:
ah yes it's not removing a specific handler I see :+1: